provider.go

```diff
@@ -1570,9 +1453,20 @@ func (p *provider) Construct(ctx context.Context, req *rpc.ConstructRequest) (*r
 	contract.Assertf(req.AcceptsOutputValues, "The caller must accept output values")
 
 	ctx = p.ctx(ctx, r.Urn)
-	result, err := p.client.Construct(ctx, r)
+	resp, err := p.client.Construct(ctx, r)
+	if err != nil {
+		return nil, err
+	}
+
+	state, err := p.asStruct(resp.State)
+	if err != nil {
+		return nil, err
+	}
 
-	return result.rpc(p.asStruct), err
+	return &rpc.ConstructResponse{
+		Urn:   string(resp.Urn),
+		State: state,
```

Comment:
If `req.AcceptsOutputValues` is false, then we need to reconstruct `StateDependencies` here.

Reply:
This implementation does not accept requests where AcceptsOutputValues is false.
```go
contract.Assertf(req.AcceptsOutputValues, "The caller must accept output values")

```

Reply:
As long as we fail loudly, I'm fine with this as is.